<launch>
  <arg name="uuv_name" />
  <arg name="start_time" default="-1"/>
  <arg name="start_x" default="0"/>
  <arg name="start_y" default="0"/>
  <arg name="start_z" default="-20"/>
  <arg name="end_x" default="50"/>
  <arg name="end_y" default="0"/>
  <arg name="end_z" default="-20"/>
  <arg name="n_points" default="50"/>
  <!-- Heading offset given in degrees -->
  <arg name="heading_offset" default="0"/>
  <arg name="duration" default="0"/>
  <arg name="max_forward_speed" default="0.3"/>

  <group ns="$(arg uuv_name)">
    <node pkg="uuv_control_utils"
      type="start_l_shape_trajectory.py"
      name="start_l_shape_trajectory"
      output="screen">
      <rosparam subst_value="true">
        start_time: $(arg start_time)
        start:
          - $(arg start_x)
          - $(arg start_y)
          - $(arg start_z)
        end:
          - $(arg end_x)
          - $(arg end_y)
          - $(arg end_z)
        n_points: $(arg n_points)
        heading_offset: $(arg heading_offset)
        duration: $(arg duration)
        max_forward_speed: $(arg max_forward_speed)
      </rosparam>
    </node>
  </group>
</launch>
